runtime/src/bank.rs

```diff
@@ -326,6 +327,7 @@ pub struct LoadAndExecuteTransactionsOutput {
     pub processed_counts: ProcessedTransactionCounts,
 }
 
+#[derive(Debug, PartialEq)]
 pub struct TransactionSimulationResult {
     pub result: Result<()>,
```

Comment:
Do you know how RPC and/or the CLI handles errors. Is it just any simulation error results in us not sending the transaction?

Reply:
Just want to make sure there's not something silly like that it checks if the transaction can be committed before sending it, rather than just error.

Basically even though these errors are now (pending feature activation) recordable/committable, we will still do right by the user and not send it to the leader.

Reply:
Yup no difference to the end user here. They should look at the `result` field here and they can't really discern between validation, processing, or execution errors.